src/mds/objects.py

```diff
@@ -226,6 +226,161 @@ async def create_object_for_id(
+        "Also used with page to determine page size.",
+    ),
+    filter: str = Query(
+        "",
+        description="The filter(s) that will be applied to the "
+        "result (more detail in the docstring).",
+    ),
+) -> JSONResponse:
+    """
+    The filtering functionality was primarily driven by the requirement that a
+    user be able to get all objects having an authz resource matching a
+    user-supplied pattern at any index in the "_resource_paths" array.
+
+    For example, given the following metadata objects:
+
+        {
+            "0": {
+                "message": "hello",
+                "_uploader_id": "100",
+                "_resource_paths": [
+                    "/programs/a",
+                    "/programs/b"
+                ],
+                "pet": "dog",
+                "pet_age": 1
+            },
+            "1": {
+                "message": "greetings",
+                "_uploader_id": "101",
+                "_resource_paths": [
+                    "/open",
+                    "/programs/c/projects/a"
+                ],
+                "pet": "ferret",
+                "pet_age": 5,
+                "sport": "soccer"
+            },
+            "2": {
+                "message": "morning",
+                "_uploader_id": "102",
+                "_resource_paths": [
+                    "/programs/d",
+                    "/programs/e"
+                ],
+                "counts": [42, 42, 42],
+                "pet": "ferret",
+                "pet_age": 10,
+                "sport": "soccer"
+            },
+            "3": {
+                "message": "evening",
+                "_uploader_id": "103",
+                "_resource_paths": [
+                    "/programs/f/projects/a",
+                    "/admin"
+                ],
+                "counts": [1, 3, 5],
+                "pet": "ferret",
+                "pet_age": 15,
+                "sport": "basketball"
+            }
+        }
+
+    how do we design a filtering interface that allows the user to get all
+    objects having an authz string matching the pattern
+    "/programs/%/projects/%" at any index in its "_resource_paths" array? (%
+    has been used as the wildcard so far because that's what Postgres uses as
+    the wildcard for LIKE) In this case, the "1" and "3" objects should be
+    returned.
+
+    The filter syntax that was arrived at ending up following the syntax
+    specified by a [Node JS implementation](https://www.npmjs.com/package/json-api#filtering) of the [JSON:API
+    specification](https://jsonapi.org/).
+
+    The format for this syntax is filter=(field_name,operator,value), in which
+    the field_name is a json key without quotes, operator is one of :eq, :ne,
+    :gt, :gte, :lt, :lte, :like, :all, :any (see operators dict), and value is
+    a typed json value against which the operator is run.
+
+    Examples:
+
+        GET /objects?filter=(message,:eq,"morning") returns "2"
```

Comment:
I was reading through the [design doc](https://docs.google.com/document/d/1vRrM5xvTepzRqw3tQpqR8hLg4QdbP3eD44_bKZsSL04/edit#) for PXP-5529, but I was curious about the query operator formats here. Is this based on a standardized format? I wonder if we need all these filter options at this stage or if it's easier to start with what we know we need and grow from there.

Reply:
I see now you have a link literally just above this regarding the format/standard as coming from JSON:API. Sorry! 😬